arch/arm/include/armv7-a/irq.h

```diff
@@ -490,13 +484,36 @@ static inline_function uint32_t up_getsp(void)
   return sp;
 }
 
-#endif /* __ASSEMBLY__ */
+static inline_function uint32_t *up_current_regs(void)
+{
+  uint32_t *regs;
+  __asm__ __volatile__
+  (
+    "mrc " "p15, " "0" ", %0, " "c13" ", " "c0" ", " "4" "\n"
```

Comment:
CP15_TPIDRPRW

Reply:
cp15.h is internal head file
we can't include cp15.h into irq.h

Reply:
let us move cp15.h to arch public, same level as armv*/

Reply:
but cp15 is only useable for R and A profile, it isn't good to bring the whole file to public folder.

Reply:
let's add  a comment